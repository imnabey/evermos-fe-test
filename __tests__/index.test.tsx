import { render, screen } from '@testing-library/react';
import Home from '@/pages/index';
import { Campaign } from '../types';

let campaigns: Campaign[] = [
  {
    data: {},
    campaign: {
      title: '#BisaBangkit Bersama Kitabisa',
      donation_received: 178613497,
      days_remaining: 0,
      donation_percentage: 0.357227,
      donation_target: 0.002,
    },
    days_remaining: 0,
    donation_target: 0.002,
  },
];

describe('[Test Suite] <Home/> Components', () => {

  it('[Test] Render <App/> element and show text "Kitabisa.com"', () => {
    render(<Home campaigns={campaigns} />);

    const result = screen.getByText(/Kitabisa.com/i);
    expect(result).toBeInTheDocument();
  });

  it('Home page snapshot testing', () => {
    const { container } = render(<Home campaigns={campaigns} />);
    expect(container).toMatchSnapshot();
  })

});