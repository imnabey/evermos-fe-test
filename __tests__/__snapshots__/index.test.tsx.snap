// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Test Suite] <Home/> Components Home page snapshot testing 1`] = `
<div>
  <main
    class="home"
  >
    <div
      class="home__container container"
    >
      <div
        class="mb-30 align-items-center row row-cols-md-2 row-cols-1"
      >
        <div
          class="d-flex align-items-center col"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2750%27%20height=%2750%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Picture of the author"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <h1
            class="fw-bold"
          >
            Kitabisa.com
          </h1>
        </div>
        <div
          class="d-flex align-items-center home__sorting col"
        >
          <div
            class="home__sorting-text fs-base mr-10"
          >
            Sorted By:
          </div>
          <div
            class="home__sorting-select"
          >
            <select
              class="form-select--grey fs-base form-select"
            >
              <option
                value="asc-donation"
              >
                Donation Goal
              </option>
              <option
                value="asc-day"
              >
                Day Left
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        class="home__item-wrapper border-radius-lg"
      >
        <div
          class="d-flex mb-20 align-items-center"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-box mr-10 fc-medium-blue fs-base"
            data-icon="box"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M50.73 58.53C58.86 42.27 75.48 32 93.67 32H208V160H0L50.73 58.53zM240 160V32H354.3C372.5 32 389.1 42.27 397.3 58.53L448 160H240zM448 416C448 451.3 419.3 480 384 480H64C28.65 480 0 451.3 0 416V192H448V416z"
              fill="currentColor"
            />
          </svg>
          Menampilkan
          <strong
            class="ml-5 mr-5 fc-medium-blue"
          >
            1
          </strong>
          data campaign.
        </div>
        <div
          class="row row-cols-lg-3 row-cols-md-2 row-cols-1"
        >
          <div
            class="home__col-card-item mb-20 col"
          >
            <div
              class="card"
            >
              <div
                class="position-relative home__card-image"
              >
                <div
                  class="position-absolute home__card-image-label fs-caption fw-semibold"
                >
                  Target:
                  <span
                    class="ml-5"
                  >
                    Rp.0.002
                  </span>
                </div>
                <img
                  class="card-img-top"
                />
              </div>
              <div
                class="card-body"
              >
                <div
                  class="fs-subhead fw-semibold fc-light-grey card-title h5"
                />
                <div
                  class="home__progress mb-15 home__progress--not-achived progress"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuenow="NaN"
                    class="progress-bar progress-bar-striped"
                    role="progressbar"
                  >
                    NaN%
                  </div>
                </div>
                <p
                  class="fs-caption fc-light-grey mb-10 card-text"
                >
                  Terkumpul
                </p>
                <p
                  class="fs-title fw-bold mb-10 pb-10 card-text"
                >
                  <span />
                </p>
                <div
                  class="border-bottom-light mb-10 mt-10"
                />
                <div
                  class="d-flex align-items-center fs-caption fc-light-grey"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-clock mr-10 fc-light-grey fs-base"
                    data-icon="clock"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 512C114.6 512 0 397.4 0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256C512 397.4 397.4 512 256 512zM232 256C232 264 236 271.5 242.7 275.1L338.7 339.1C349.7 347.3 364.6 344.3 371.1 333.3C379.3 322.3 376.3 307.4 365.3 300L280 243.2V120C280 106.7 269.3 96 255.1 96C242.7 96 231.1 106.7 231.1 120L232 256z"
                      fill="currentColor"
                    />
                  </svg>
                  <p
                    class="mr-5 card-text"
                  >
                    0
                  </p>
                  <p
                    class="card-text"
                  >
                    Sisa Hari
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
